### DISCUSSION TITLE
React hook form doesn't see the changes of input when value is useState

### DISCUSSION THREAD

**Discussion by daler-sz** [react-hook-form/react-hook-form#10648] [General] [answered]
I think that [this behaviour ](https://stackoverflow.com/questions/76623016/react-hook-form-doesnt-see-the-changes-of-input-when-value-is-usestate) shoul be fixed or at least described in docs

**Comment by leapful** [comment, marked as answer]
@daler-sz 

There is no bug to fix for this behavior because it's a wrong logic handling from developer's implementation and it's impossible to update documentation to describe all the mistakes of developer in the world (the documentation will be too long to infinity).

react-hook-form or any form handling library for web browser will handle get and set HTML input value via attribute `value` and `onChange`.

The example in stackoverflow uses a custom local state to handle input value by themself and overrides it by using `<input {...register('input')} value={state} onChange={() => setState()}/>` therefore it is also their responsibility to keep syncing local state with HTML and react-hook-form.
